appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -0,0 +1,390 @@
+
+import java.util.List;
+
+public final class MockChart extends MockContainer {
+    public static final String TYPE = "Chart";
+
+    private static final String PROPERTY_NAME_TYPE = "Type";
+    private static final String PROPERTY_NAME_DESCRIPTION = "Description";
+    private static final String PROPERTY_NAME_LEGEND_ENABLED = "LegendEnabled";
+    private static final String PROPERTY_NAME_GRID_ENABLED = "GridEnabled";
+    private static final String PROPERTY_NAME_PIE_RADIUS = "PieRadius";
+    private static final String PROPERTY_NAME_LABELS_FROM_STRING = "LabelsFromString";
+
+    static {
+        ResourcesType.setClientBundle(EmbeddedResources.INSTANCE);
+    }
+
+    protected MockChartView chartView;
+
+    // Legal values for type are defined in
+    // com.google.appinventor.components.common.ComponentConstants.java.
+    private int type;
+
+    // Keep track whether the children of the Mock Chart have been
+    // reattached. The reattachment has to happen only once, since the Data
+    // Series are part of the Chart object itself.
+    private boolean childrenReattached = false;
+
+    // Store the LabelsFromStrings property parsed result
+    private String[] labelArray = new String[0];
+
+    /**
+     * Creates a new instance of a visible component.
+     *
+     * @param editor editor of source file the component belongs to
+     */
+    public MockChart(SimpleEditor editor) {
+        super(editor, TYPE, images.image(), new MockChartLayout());
+
+        // Since the Mcok Chart component is not a container in a normal
+        // sense (attached components should not be visible), the Chart Widget
+        // is added to the root panel, and the root panel itself is initialized.
+        // This is done to ensure that Mock Chart Data components can be dragged
+        // onto the Chart itself, rather than outside the Chart component.
+        rootPanel.setStylePrimaryName("ode-SimpleMockComponent");
+
+        // Since default type property does not invoke the setter,
+        // initially, the Chart's type setter should be invoked
+        // with the default value.
+        setTypeProperty("0");
+
+        initComponent(rootPanel);
+    }
+
+    @Override
+    protected void onAttach() {
+        super.onAttach();
+
+        // The Children of the Mock Chart have not yet been attached
+        // (this happens upon initializing the Chart which has child components)
+        if (!childrenReattached) {
+            // Attach all children MockComponents
+            for (MockComponent child : children) {
+                if (child instanceof MockChartData) {
+                    // Re-add Data Components to the Mock Chart
+                    ((MockChartData) child).addToChart(MockChart.this);
+                }
+            }
+
+            // Update the state of children to reattached
+            childrenReattached = true;
+        }
+    }
+
+    @Override
+    public void delete() {
```

Comment:
This is (to be) fixed in #1858 and was used as a temporary fix without modifying the MockContainer. To be removed after merging.

Reply:
#1858 was merged, so you can resolve whatever needs to be done here.